src/screens/NewsListScreen.js

```diff
@@ -157,3 +75,34 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
   },
 });
+
+const mapStateToProps = state => {
+  return {
+    isLoading: state.isLoading,
+    articles: state.articles,
+    page: state.page,
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    onRefresh: () => {
+      if (ownProps.isLoading) {
+        return;
+      }
+      dispatch(invalidateNews());
+      dispatch(loadArticles());
```

Comment:
You only need to invalidateArticles, then in `componentWillReceiveProps` and `componentDidMount` you should compare `props.valid` and `nextProps.valid` and decide to call `loadArticles`